common/src/main/java/org/apache/uniffle/common/netty/protocol/SendShuffleDataRequest.java

```diff
@@ -103,7 +103,19 @@ private static Map<Integer, List<ShuffleBlockInfo>> decodePartitionData(ByteBuf
       int lengthOfShuffleBlocks = byteBuf.readInt();
       List<ShuffleBlockInfo> shuffleBlockInfoList = Lists.newArrayList();
       for (int j = 0; j < lengthOfShuffleBlocks; j++) {
-        shuffleBlockInfoList.add(Decoders.decodeShuffleBlockInfo(byteBuf));
+        try {
+          shuffleBlockInfoList.add(Decoders.decodeShuffleBlockInfo(byteBuf));
+        } catch (Throwable t) {
+          // An OutOfDirectMemoryError will be thrown, when the direct memory reaches the limit.
+          // OutOfDirectMemoryError does not cause the jvm exit and cause the direct memory leak.
+          // Warning: Please set DIRECT_MEMORY_LIMIT to a reasonable value.
```

Comment:
`DIRECT_MEMORY_LIMIT`  does not exist? Maybe `DIRECT_MEMORY_LIMIT` -> `MAX_DIRECT_MEMORY_SIZE`?

Reply:
You can also put the link of the [guidance](https://github.com/apache/incubator-uniffle/pull/1893) in the comment.